include/ada/implementation.h

```diff
@@ -34,7 +34,12 @@ namespace ada {
   ada_warn_unused ada::url parse(std::string_view input,
                                  const ada::url* base_url = nullptr,
                                  ada::encoding_type encoding = ada::encoding_type::UTF8);
-
+  /**
+   * Computes a href string from a file path.
+   * @param input a path
+   * @return a href string (starts with file:://)
+   */
+  std::string href_from_file(std::string_view path);
```

Comment:
shouldn't this be `const`?

Reply:
It is a free function (not part of `url`) so `const` is not relevant. You would call is as `mystring = ada::href_from_file("/home/non/text.txt")`.

It is a rather large function so it seems unwise to inline it. It only looks trivial.